### PR TITLE
doc: Refactors the documentation

### PR DESCRIPTION
**Pull request by jniggemann**
This commit refactors the documentation according to my proposal in #1273
and the discussion I had with fd0 on IRC.

The bits from the manual that I could not immediately put into the new
structure are contained in manual_rest.rst Anything else is still there,
nothing has been deleted.

I changed the heading markup to follow the convention used in Python’s
Style Guide for documentation, this convention is explained in a comment
at the top of every file.

I also added a paragraph on installing restic on Debian.

### CODE REVIEW

**Comment by jniggemann** [comment]
I incorporated all commits relating to documentation that were created after creating my branch.

**Comment by fd0** [comment]
Looks good!

File: doc/manual_rest.rst [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+.. code-block:: console
+
+    $ export TMPDIR=/var/tmp/restic-tmp
+    $ restic -r /tmp/backup backup ~/work
+
+
+
+Caching
+-------
```
Comment (prattmic) [inline doc/manual_rest.rst]: FYI in case my PR goes in before yours, #1315 is changing this section.

File: doc/100_references.rst [hunk 2]
```diff
@@ -607,3 +624,140 @@ stored files) which files belong to what snapshot. When only these files
...
+Local Cache
+===========
+
+In order to speed up certain operations, restic manages a local cache of data.
+This document describes the data structures for the local cache with version 1.
+
+Versions
+--------
+
+The cache directory is selected according to the `XDG base dir specification
```
Comment (prattmic) [inline doc/100_references.rst]: This isn't quite right anymore, the XDG spec is only used on Linux (actually anything non-windows, non-darwin).

See also #1315

**Comment by fd0** [comment]
I've taken the liberty of rebasing the branch onto master and pushed it. I'll merge it once the CI tests are happy.

**Comment by fd0** [comment]
merged, thanks!

**Pull request merged**